Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -143,19 +159,17 @@
           "Selector": "$.SnapshotId",
           "Type": "String"
         }
-      ],
-      "name": "copyAndEncryptSnapshot",
-      "action": "aws:executeAwsApi",
-      "timeoutSeconds": 3600,
-      "onFailure": "Abort"
+      ]
     },
+
     {
       "name": "waitForSnapshot",
       "action": "aws:waitForAwsResourceProperty",
-      "timeoutSeconds": "180",
+      "timeoutSeconds": 1800,
```

Comment:
Shouldn't this be {{ maxWaitForSnapshotWithEncryption }} ?

Reply:
I missed your comment on this. It sounds like a bug, but try to change the maxWaitForCreateSnapshot parameter type to String (even though it is a number) and see if that works.

If that is the case, you can add a regEx to the String parameter that represents the timeout value such as:

[0-9]{1,5}

While not perfect, it will allow for up to 99999 seconds approx 27 hours, which is massive.

Reply:
Yes I think it is definitely a bug and more specifically, I think the Automation Document is having issues with the fact that a String has been passed in for "timeoutSeconds".  I've done some testing and while I could get "allowedPattern": "[0-9]{5}" to pass for the parameter, when I try to use the value, it still says malformed JSON and point to the same line.  So I am thinking of just reverting back to Integer and file a bug (whom do I talk to?  Or should I just open an Issue on this repo?)

Reply:
For now, let's revert to the previous version without the input parameters for the timeout, and statically set 30 minutes for the maximum wait for the wait step.

Reply:
OK got it.  It's in the latest Commit.